### DISCUSSION TITLE
How to delete ents in a doc

### DISCUSSION THREAD

**Discussion by Alvaro8gb** [explosion/spaCy#11853] [Help: Coding & Implementations] [answered]
I am trying to delete an entity but its giving me errors, with this code
`doc[i].ent_type_ = ""`
Can anyone help me ? :)

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @Alvaro8gb , you have the following options here:

- Get all the entities with `doc.ents`, remove what you don't need, and set it back again. 

```python
entities = doc.ents
new_entities = delete_entities(entities)  # implement this
doc.ents = new_entities
```

- The [`doc.set_ents` function](https://spacy.io/api/doc#set_ents) also has parameters for setting entities as `O` (as in outside in IOB) like `doc.set_ents(outside=[ent])` or you can set some as missing, like `doc.set_ents(missing=[ent])`.

**Reply by Alvaro8gb** [reply]
Thanks @ljvmiranda921, I solved it like this 2º Forms:

`new_entities = []`
`new_entities.append(Span(doc, start, end, label=id))`
`doc.ents = new_entities`

or

`doc.set_ents([Span(doc, start, end, id)])`

These is very useful if someone want to empower a NER model with regex recognition.
